### PR TITLE
Add 'addToRequestQueryString' helper function

### PR DESCRIPTION
**Pull request by bsima**
This is a helper function that I found useful and extracted from a private codebase during refactoring. Submitting it here for possible inclusion. Let me know if you need anything else before merging.

### CODE REVIEW

File: http-conduit/Network/HTTP/Simple.hs [hunk 1]
```diff
@@ -347,6 +348,13 @@ getRequestQueryString = H.parseQuery . H.queryString
 setRequestQueryString :: H.Query -> H.Request -> H.Request
 setRequestQueryString = H.setQueryString
 
+-- | Add to the existing query string parameters.
+--
+-- @since 2.3.5
```
Comment (snoyberg) [inline http-conduit/Network/HTTP/Simple.hs]: Overall looks good. Can you add a cabal file version bump and a ChangeLog entry too?

**Comment by bsima** [comment]
Michael Snoyman <notifications@github.com> writes:

> Overall looks good. Can you add a cabal file version bump and a ChangeLog entry too?

Fixed in 99ea17a0088e4a8188e5e6a9a88d2ea2eac0032f, lemme know if you
need squashing/rebasing.

**Comment by snoyberg** [comment]
I don't see that commit here, maybe you didn't push yet?

**Comment by bsima** [comment]
Michael Snoyman <notifications@github.com> writes:

> I don't see that commit here, maybe you didn't push yet?

Sorry, push failed and I walked away without noticing. It's there now!

**Pull request merged**